syntax = "proto3";

package masterworker;

service AssignTask {
    rpc Ping(PingRequest) returns (TaskReply){}
    rpc Map(MapRequest) returns (TaskReply){}
    rpc Reduce(ReduceRequest) returns (TaskReply){}
}

message PingRequest{
    //more stuff as we need it
}

message MapRequest{
    string user_id = 1;
    string job_id = 2;
    repeated ShardPiece shard = 3;
    // other stuff as we need it
}

message ShardPiece{
    string file_name = 1;
    int32 start_index = 2;
    int32 end_index = 3;
}

message TaskReply{
    //the job id of the task, set only for map and reduce
    string job_id = 1;
    //the file that was written to
    string out_file = 2;

    // other stuff as we need it

    //the type of task that is returned
    string task_type = 3;
}

message ReduceRequest{
    string user_id = 1;
    string output_file = 2;
    string job_id = 3;
}
